frontend/src/screens/learner.tsx

```diff
@@ -1,49 +1,88 @@
-import { React, useState } from '@common'
-import { ParticipantStudy } from '@api'
-import { Footer, TopNavBar } from '@components'
-import { useEnvironment, useIsMobileDevice } from '@lib'
-import { useParticipantStudies, useSearchStudies } from './learner/studies'
-import { StudyCard } from './learner/card'
-import { StudyDetails } from './learner/details'
-import { Route, Routes } from 'react-router-dom'
+import { React } from '@common';
+import { FC, useMemo, ReactNode } from 'react';
+import { ParticipantStudy } from '@api';
+import { Footer, TopNavBar } from '@components';
+import { useEnvironment, useIsMobileDevice } from '@lib';
+import { useParticipantStudies, useSearchStudies } from './learner/studies';
+import { StudyCard } from './learner/card';
+import { StudyDetails } from './learner/details';
+import { Route, Routes } from 'react-router-dom';
 import { Swiper, SwiperSlide } from 'swiper/react';
-import { EffectCards, FreeMode, Navigation, Pagination } from 'swiper/modules';
+import { EffectCards, FreeMode } from 'swiper/modules';
 import { LearnerWelcomeModal } from './learner/learner-welcome-modal';
 import { UnsupportedCountryModal } from './learner/unsupported-country-modal';
-import { Badge, Box, Container, Flex, Group, Stack, Text, TextInput, Title } from '@mantine/core';
+import {
+    Badge,
+    Box,
+    Container,
+    Divider,
+    Flex,
+    Group,
+    Stack,
+    Text,
+    TextInput,
+    Title,
+} from '@mantine/core';
 import { IconSearch, IconX } from '@tabler/icons-react';
-import { groupBy, filter } from 'lodash';
-import { colors } from '@theme'
-import { useMemo } from 'react';
-import { orderBy, sortBy, uniqBy } from 'lodash-es';
+import { groupBy, orderBy, uniqBy } from 'lodash-es';
+import 'swiper/css';
+import 'swiper/css/effect-cards';
+import 'swiper/css/free-mode';
+import 'swiper/css/navigation';
+import { colors } from '@theme';
+import { useWindowScroll } from '@mantine/hooks';
+import { Button } from '@mantine/core';
+import { IconArrowUp } from '@tabler/icons-react';
+
+
+const sortAndMapStudies = (
+    studies: ParticipantStudy[],
+    renderStudy: (study: ParticipantStudy) => ReactNode
```

Comment:
i think we could just return the JSX here from the template rather than passing a method to render the jsx? i think that would be simpler

like calling this method `renderStudies`

```
    return studies
        .sort((a, b) =>
            a.completedAt === b.completedAt
                ? 0
                : a.completedAt
                    ? 1
                    : -1
        )
        .map((study) => {
          return (
            <SwiperSlide style={{ padding: '1rem' }} key={study.id}>
              <StudyCard study={study} />
            </SwiperSlide>
          )
        });
```

and later in the jsx? 

```
{renderStudies(studies)}
```

Reply:
I think it can be furthur simplied by implementing the sort function in the useSearchStudies hook.